File: scripts/heater.js
const heater = extendContent(GenericCrafter, "heater", {
	load() {
		this.super$load();
		this.regions = [];
		this.regions[0] = Core.atlas.find(this.name);
		this.regions[1] = Core.atlas.find(this.name + "-liquid");
		this.regions[2] = Core.atlas.find(this.name + "-top");
	}
});

heater.buildType = () => extendContent(GenericCrafter.GenericCrafterBuild, heater, {
	draw() {
	    Draw.rect(heater.regions[0], this.x, this.y);
	    //Drawf.liquid(heater.regions[1], this.x, this.y, this.liquids.get(outputLiquid.liquid) / this.liquidCapacity, this.outputLiquid.color);
	    Draw.rect(heater.regions[2], this.x, this.y);
	}
});

File: src/revision/world/blocks/production/Heater.kt
package revision.world.blocks.production

import arc.Core
import arc.graphics.g2d.Draw
import arc.graphics.g2d.TextureRegion
import mindustry.content.Liquids
import mindustry.graphics.Drawf
import mindustry.world.blocks.production.GenericCrafter

open class Heater(name: String) : GenericCrafter(name) {

    var liquidRegion: TextureRegion? = null
    var topRegion: TextureRegion? = null

    override fun load() {
        super.load()
        liquidRegion = Core.atlas.find("$name-liquid")
        topRegion = Core.atlas.find("$name-top")
    }

    inner class HeaterBuild : GenericCrafterBuild() {

        override fun draw() {
            super.draw()
            Drawf.liquid(liquidRegion, x, y, liquids.get(outputLiquid.liquid) / liquidCapacity, Liquids.water.color)
            Draw.rect(topRegion, x, y)
        }

    }
}